tools/pylib/boutdata/data.py

```diff
@@ -247,6 +249,50 @@ def __init__(self, filename, name="root"):
 
                         section[line[:eqpos].strip()] = value
 
+    def write_new(self, filename=None, force_overwrite=False):
+        """
+        Write a new input file with the options (including any modifications
+        since reading in).
+        By default this will overwrite the original input file, prompting the
+        user for confirmation.
+        Note, no comments from the original input file are transferred to the
+        new one.
+
+        Parameters
+        ----------
+
+        filename : str
+            Path of the file to write
+            (defaults to path of the file that was read in)
+        force_overwrite : bool
+            If False then prompt the user whether to overwrite an existing
+            file, if there is one. Otherwise, just overwrite without asking.
+            (default False)
+        """
+        if filename is None:
+            filename = self.filename
+
+        if not force_overwrite and os.path.exists(filename):
+            user_input = "a"
+            while user_input not in ["y", "n", ""]:
+                user_input = input(filename+" already exists: do you want to over-write it? y/[n]")
```

Comment:
I'm not sure about this. It's fine if it's being used interactively, but in a script, this could cause the process to just wait indefinitely. Probably better to just raise the `ValueError` instead